src/fheroes2/castle/castle_mageguild.cpp

```diff
@@ -132,11 +177,36 @@ void Castle::_openMageGuild( const Heroes * hero ) const
 
         buttonExit.drawOnState( le.isMouseLeftButtonPressedAndHeldInArea( buttonExit.area() ) );
 
+        if ( buttonPrevCastle.isEnabled() ) {
+            buttonPrevCastle.drawOnState( le.isMouseLeftButtonPressedAndHeldInArea( buttonPrevCastle.area() ) );
+
+            if ( le.MouseClickLeft( buttonPrevCastle.area() ) || HotKeyPressEvent( Game::HotKeyEvent::DEFAULT_LEFT ) || timedButtonPrevCastle.isDelayPassed() ) {
+                return MageGuildDialogResult::PrevMageGuildWindow;
+            }
+        }
+        if ( buttonNextCastle.isEnabled() ) {
+            buttonNextCastle.drawOnState( le.isMouseLeftButtonPressedAndHeldInArea( buttonNextCastle.area() ) );
+
+            if ( le.MouseClickLeft( buttonNextCastle.area() ) || HotKeyPressEvent( Game::HotKeyEvent::DEFAULT_RIGHT ) || timedButtonNextCastle.isDelayPassed() ) {
+                return MageGuildDialogResult::NextMageGuildWindow;
+            }
+        }
+
         spellRows[0]->queueEventProcessing( false ) || spellRows[1]->queueEventProcessing( false ) || spellRows[2]->queueEventProcessing( false )
             || spellRows[3]->queueEventProcessing( false ) || spellRows[4]->queueEventProcessing( false );
 
         if ( le.isMouseRightButtonPressedInArea( buttonExit.area() ) ) {
             fheroes2::showStandardTextMessage( _( "Exit" ), _( "Exit this menu." ), Dialog::ZERO );
```

Comment:
What do you think about making this text exact the same as in line 150:

```suggestion
            fheroes2::showStandardTextMessage( _( "Exit" ), _( "Exit Mage Guild" ), Dialog::ZERO );
```
To be consistent with the status bar messages and also it looks better because we did not open any menu prevoiusly. :)